packages/page-contracts/src/Codes/Upload.tsx

```diff
@@ -202,7 +202,7 @@ function Upload ({ onClose }: Props): React.ReactElement {
             <InputBalance
               help={t<string>('The balance to transfer from the `origin` to the newly created contract.')}
               isError={!isValueValid}
-              isZeroable={hasStorageDeposit}
+              isZeroable={contractAbi.constructors[constructorIndex].isPayable}
```

Comment:
What does `isZeroable` do if true?

Reply:
it makes the input validation fail. the input gets a red border and you can't move to the next step

Reply:
now i tried with the older node and even if the constructor is not payable, the transaction fails with "contract not founded" if i pass 0.
is is safe to assume that if the node supports storage deposit than the value can be 0 for non payable constructors?

Reply:
You have to differentiate between ABI (payable) and the node. ABI comes form ink! and is decoupled from the node version.

Old Node:
Always needs `value/endowment` in order to pay for rent. Disregard `payable`.

New Node:
Does not need any `value` **except** the constructor is `payable`. Hence you should not even show the field and use `0` when the constructor isn't payable.

ink!:
Ink constructors are implicitly **payable***. The fuck up is that this is implicit. The field is missing from ABI. What you should do is interpete the absense of the payable field as `payable = true`. We intent to make it explicit and also allow to make them not payable. The following is forward compatible:
`undefined` -> true
`false`-> false
`true` -> true


The

Reply:
ok, got it! 
i am using [this](https://github.com/polkadot-js/apps/blob/b3ad01d90dd27ba0d1b3a81fc07f40f6e6aa8566/packages/page-contracts/src/Contracts/Call.tsx#L49) to check if the node has storage deposit or not. because the changes to value are related to storage deposit i wanted to double check it the same condition can be used